API.md

```diff
@@ -0,0 +1,42 @@
+
+### `SegM8(uint8_t pinCS, uint8_t devicesCount = 1)`
+
+Constructs a new SegM8 object that uses the default hardware SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin used to control the shift-register latch. It is marked `CS` on the SegM8 board. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+### `SegM8(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1)`
+
+Constructs a new object of data bus Software SPI with arguments:
+
+- `pinCS`: the chip select (also known as slave select) pin used to control the shift-register latch. It is marked `CS` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinMOSI`: the MOSI (Master Out Slave In) pin used to send data to the peripherals. It is marked `DI` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinSCK`: the clock pin used to clock pulses which synchronize data transmission generated by the master. It is marked `CLK` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+Difference between hardware and software SPI is speed. Software SPI runs slower than hardware SPI but hardware SPI works only on specific board pins: `MOSI`, `MISO` (not used in SegM8) and `SCK`.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for communication. All module segments are set to the 'dark' state.
+
+Call this method before interfacing with SegM8. For example, in your `setup()`.
+
+### `void printf(const char* format, ...)`
+
+Displays a 'c'-formatted string on a chain of `SegM8` indicator modules. The format string syntax matches the standard C `printf` function. For example: `Clock.printf("%2d:%2d",hours,minutes);`
+The 7-segment indicator output supports digits, excluding floating-point, and limited support for English characters except of `K,M,N,V,W,X,Z`. Note that the display of letters on a 7-segment indicator does not always correspond to their generally accepted spelling.
+Also supported are the characters: space, underscore, dot, minus, degree (at `^`). Unsupported characters are displayed as `blank space`.
+**Important: all Arduino xprintf functions does not support `%f` format specifier correctly, do the conversion float/double to a string yourself and print the result as a string.**
+
+### `void setSegments(uint8_t device, uint8_t segments)`
+
+Display not ordinary symbol to the specified position, or control some special modules (for example [Octofet board](https://my.amperka.com/modules/octofet)) in chain. You can use special modules for emulate unsupported symbols: colon, semi-colon, plus, asterisk for example, using plane leds.
```

Comment:
```suggestion
Displays a special symbol in the specified position. To emulate an unsupported symbol: colon, semi-colon, plus, asterisk special modules may be used e.g. plane LEDs. It can be also used to control some special modules (for example [Octofet board](https://my.amperka.com/modules/octofet)) in a daisy-chain.
```

Reply:
Matthew, it is not right. Colon, semicolon, etc we can emulate only using octofet. It must be after link to octofet.

Reply:
I don't understand. Why only octofet? Here is an initial sentence: «You can use special modules for emulate unsupported symbols: colon, semi-colon, plus, asterisk for example, using plane leds.» It says that we can emulate theese symbols with LEDS.

Reply:
Plane LEDs do not have SPI interface :) For controlling of LEDs we need some special devices - octofet for examle